controllers/webspherelibertyapplication_controller.go

```diff
@@ -535,6 +508,35 @@ func (r *ReconcileWebSphereLiberty) Reconcile(ctx context.Context, request ctrl.
 
 	}
 
+	// Check if SemeruCloudCompiler is enabled before reconciling the Semeru Compiler deployment and service.
```

Comment:
Semeru Cloud Compiler should be up and running prior to creating application deployment/statefulset. Suggest to move up, after [this](https://github.com/WASdev/websphere-liberty-operator/blob/b3279581cf450991ed6f408377d97f72fb311dc4/controllers/webspherelibertyapplication_controller.go#L267) now that we rely on serviceaccount.